server/market/market.go

```diff
@@ -249,37 +303,92 @@ ordersLoop:
 			continue ordersLoop
 		}
 
-		// All coins are unspent. Lock them.
-		if lo.Sell {
+		if baseIsAcctBased {
+			var addr string
+			var qty, lots uint64
+			var redeems int
+			if lo.Sell {
+				// address is zeroth coin
+				if len(lo.Coins) != 1 {
+					log.Errorf("rejecting account-based-base-asset order %s that has no coins ¯\\_(ツ)_/¯", lo.ID())
+					continue ordersLoop
+				}
+				addr = string(lo.Coins[0])
```

Comment:
This would be an encoded `AmountCoinID`, not just an address.

Reply:
As we wiggle everything into place, one of the things we need to do is to "minimize" the API by eliminating any extraneous information. The server doesn't need the value encoded in the coin ID. It's knows how much to lock for the order.

Just looking forward, I would expect a number of changes to the existing server-side coin ID system. I'll comment in more detail on #1154 today so we can plan it out together. I sort of feel bad for not being more involved with the details of the coin IDs earlier, but it was so early in the process and I hadn't quite wrapped my head around everything. I'm getting there now.